### ISSUE TITLE
Is it possible to manually cancel the hook execution?

### ISSUE THREAD

**Issue by dj-neza** [simoneb/axios-hooks#376]
Hi!
I'm wondering if there's a way to cancel the hook execution manually. In some cases I need to run the execute function within the useEffect hook based on some state changes and am getting the React warning if the component unmounts so I'd like to do the cleanup properly. I found examples to do this when using Axios requests but don't know how to approach it when using this hook.

**Comment by simoneb** [comment]
It is currently not possible but it should be fairly easy to do. If you would like to send a PR the change should only be a matter of changing this line https://github.com/simoneb/axios-hooks/blob/20ebb993fa778b6000f1ef203bfc14225b64246b/src/index.js#L267

to return an additional array element which is the `cancelOutstandingRequest` function. It would need also some tests.

**Comment by dj-neza** [comment]
@simoneb Sure I can do that. Not super sure what to do about testing though as I have no idea how to test this with the testing-library

**Comment by simoneb** [comment]
> @simoneb Sure I can do that. Not super sure what to do about testing though as I have no idea how to test this with the testing-library

you'll find plenty of test cases in index.test.jsx

**Comment by dj-neza** [comment]
Ok will try to put something together in the upcoming days :)

**Comment by simoneb** [comment]
Sure, let me know if you need any input. The production code change itself is very simple.

**Comment by vmlopezr** [comment]
Hi everyone, i'm interested in getting this implemented as well. And have the following changes in my [fork](https://github.com/vmlopezr/axios-hooks/tree/return-cancel-outstanding-request)

```jsx
// index.js
   return [state, refetch, cancelOutstandingRequest];

// Tests in the cancellation and manual refetch sections
 it('should cancel the outstanding request when the cancel method is called', async () => {
     axios.mockResolvedValue({ data: 'whatever' })
     const { waitForNextUpdate, result } = setup()
     await waitForNextUpdate()
     result.current[2]()
     expect(cancel).toHaveBeenCalled()
})
it('should cancel the outstanding request when the cancel method is called after the component rerenders with same config', async () => {
    axios.mockResolvedValue({ data: 'whatever' })
    const { waitForNextUpdate, rerender, result } = setup('initial config')
    await waitForNextUpdate()
    rerender()
    result.current[2]()
    expect(cancel).toHaveBeenCalled()
})
it('should cancel the outstanding manual refetch when the cancel method is called', async () => {
    axios.mockResolvedValue({ data: 'whatever' })
    const { result, waitForNextUpdate } = setup('', { manual: true })
    act(() => {
      result.current[1]()
    })
    await waitForNextUpdate()
    result.current[2]()
    expect(cancel).toHaveBeenCalled()
})
```
Let me know if this is along the lines of what you were thinking. If so I can submit a PR.

**Comment by simoneb** [comment]
please submit a PR, it's the best way to discuss

**Comment by simoneb** [comment]
Fixed in #378
